src/ring/rack.clj

```diff
@@ -25,8 +25,9 @@
 
       def responsify(output)
         begin
-          return output[0].to_java if output.respond_to?(:size) && output.size > 1
-          output.each{|s| s.to_java}
+          retval = []
+          output.each {|s| retval.push(s)}
+          retval
```

Comment:
Ok, so line 28 is a typo and should be `>=`?

Anyways, this code is empirical also, with the original code only the frontpage works and nothing else (all the bodies are a wrong type). Now they all work.